File: week07/explore_phila_data/extract_phl_pwd_parcels.mjs
import dotenv from 'dotenv';
dotenv.config();

import fetch from 'node-fetch';
import fs from 'fs/promises';
import path from 'path';
import { fileURLToPath } from 'url';
import { Storage } from '@google-cloud/storage';

const __dirname = path.dirname(fileURLToPath(import.meta.url));
// const __dirname = new URL('.', import.meta.url).pathname; // not working because there is a leading slash
const DATA_DIR = __dirname + '/raw_data';
// create the directory if it doesn't exist
await fs.mkdir(DATA_DIR, { recursive: true });

const url = 'https://opendata.arcgis.com/datasets/84baed491de44f539889f2af178ad85c_0.geojson';
const filename = path.join(DATA_DIR, 'phl_pwd_parcels.geojson');

const response = await fetch(url);
if (!response.ok) {
  throw new Error(`HTTP error! status: ${response.status}`);
}

await fs.writeFile(filename, await response.text());

console.log(`Downloaded ${filename}`);

// Upload the downloaded file to cloud storage
const bucketName = process.env.DATA_LAKE_BUCKET;
const blobName = 'raw/phl_pwd_parcels/phl_pwd_parcels.geojson';

const storageClient = new Storage();
const bucket = storageClient.bucket(bucketName);
await bucket.upload(filename, {
  destination: blobName,
});

console.log(`Uploaded ${blobName} to ${bucketName}`);

File: week07/explore_phila_data/extract_phl_pwd_parcels.py
from dotenv import load_dotenv
load_dotenv()

import pathlib
import requests
from google.cloud import storage
import os

DATA_DIR = pathlib.Path(__file__).parent / 'raw_data'
DATA_DIR.mkdir(exist_ok=True, parents=True)  # Create the directory if it doesn't exist

# Download the OPA Properties data as a CSV
url = 'https://opendata.arcgis.com/datasets/84baed491de44f539889f2af178ad85c_0.geojson'
filename = DATA_DIR / 'phl_pwd_parcels.geojson'

response = requests.get(url)
response.raise_for_status()

with open(filename, 'wb') as f:
    f.write(response.content)

print(f'Downloaded {filename}')


# Upload the downloaded file to cloud storage
bucket_name = os.getenv('DATA_LAKE_BUCKET')
blobname = 'raw/phl_pwd_parcels/phl_pwd_parcels.geojson'

storage_client = storage.Client()
bucket = storage_client.bucket(bucket_name)
blob = bucket.blob(blobname)
blob.upload_from_filename(filename)

print(f'Uploaded {blobname} to {bucket_name}')